components/benefit_sponsors/app/models/benefit_sponsors/exporters/benefit_application_issuers.rb

```diff
@@ -0,0 +1,30 @@
+module BenefitSponsors
+  module Exporters
+    class BenefitApplicationIssuers #Exporter
+      attr_reader :lines
+
+      def retrieve(compare_date = TimeKeeper.date_of_record)
+        BenefitSponsors::BenefitSponsorships::BenefitSponsorship.where(:benefit_applications => {
+          :$elemMatch => {:"effective_period.max".gt => compare_date, :aasm_state => :active }
+        }).each do |benefit_sponsorship|
+          benefit_sponsorship.benefit_applications.each do |benefit_application|
+            @lines += BenefitSponsors::Serializers::BenefitApplicationIssuer.to_csv(benefit_application)
+          end
+        end
+      end
```

Comment:
I don't think this query fully captures the right data set.  Remember we want currently active, new but published to glue, renewing but published to GlueDB benefit applications.

Get Ram to guide you through this.

I would also recommend you create a spec example just for the query to that captures those cases above that we care about.

Reply:
Thanks for point this out, you're right, and it all depends on whether we're past the `aca_shop_market_employer_transmission_day_of_month` or not. So I moved that part of the query to it's own method and if it's past the transmission day we will look for those have start this month but haven't been transmitted yet, which we can tell from the `:aasm_state` of `:enrollment_eligible`.